### PR TITLE
Merge upstream changes

### PR DESCRIPTION
**Pull request by ClearlyClaire**
## Conflicts

- `.github/workflows/build-image.yml`:
  Upstream updated `docker/build-push-action`, and we a different config for `docker/metadata-action` so the lines directly above were different, but it's not a real conflict.
  Upgraded `docker/build-push-action` as upstream did.
- `app/javascript/mastodon/features/compose/components/compose_form.js`:
  Upstream changed the codestyle near a line we had modified to accommodate configurable character count.
  Kept our change.

## Ported changes

- 2a4f2216d65bd5ac90239d5b99c70f4330d76fde
- 9cdd643564ef1f885a4c501ac0dfc437291466a7
- d9088ef3272421a9267467fb95674d4b4afb38ab
- c49213f0ea311daba590db1d7a14a641cbd9fe93 (kinda, as 155424e52f2d382ccaf4560b2d62aba347a8992b)
- 131e6403ccd59693e8498d175acba4fdd28268b9
- e73b55184b37e7f29e5bcb16c980dc8dbdb9a7d1

### CODE REVIEW

File: app/javascript/flavours/glitch/utils/hashtag.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 export function recoverHashtags (recognizedTags, text) {
   return recognizedTags.map(tag => {
-      const re = new RegExp(`(?:^|[^\/\)\w])#(${tag.name})`, 'i');
-      const matched_hashtag = text.match(re);
-      return matched_hashtag ? matched_hashtag[1] : null;
-    }
+    const re = new RegExp(`(?:^|[^\/)\w])#(${tag.name})`, 'i');
```
[the remark this answers is not in the document]
Comment (ClearlyClaire) [reply app/javascript/flavours/glitch/utils/hashtag.js:3]: CodeQL points at an actual issue with that part of the code, but it is not new.

**Review by Plastikmensch — COMMENTED** [review]
I have some minor and some very minor complains about the style fixes.

File: app/javascript/flavours/glitch/reducers/meta.js [hunk 2]
```diff
@@ -13,14 +13,10 @@ const initialState = ImmutableMap({
 export default function meta(state = initialState, action) {
   switch(action.type) {
   case STORE_HYDRATE:
-    return state.merge(
-      action.state.get('meta'))
-        .set('permissions', action.state.getIn(['role', 'permissions']))
-        .set('layout', layoutFromWindow(action.state.getIn(['local_settings', 'layout']))
-      );
+    return state.merge(action.state.get('meta')).set('permissions', action.state.getIn(['role', 'permissions'])).set('layout', layoutFromWindow(action.state.getIn(['local_settings', 'layout'])));
```
Comment (Plastikmensch) [inline app/javascript/flavours/glitch/reducers/meta.js:16]: I see that this abides to style rules, but I don't particularly find this easy to read.
Upon further inspection, I don't like this return at all.
I initially thought about changing the indent rule, but it wouldn't help in this case at all.
The `);` confused me.

Maybe something like this, which both maintains readability while abiding to style rules:

```suggestion
    return state.merge(action.state.get('meta'))
      .set('permissions', action.state.getIn(['role', 'permissions']))
      .set('layout', layoutFromWindow(action.state.getIn(['local_settings', 'layout'])))
    ;
```

I don't particularly like the standalone semi there either, but it communicates the return statement is ending there. (Alternatively the semi can also be set after the second .set)

**Review by Plastikmensch — APPROVED** [review]
Thanks! :3

**Pull request merged**